src/controller/model/endpoint.ts

```diff
@@ -573,6 +573,33 @@ class Endpoint extends Entity {
         }
     }
 
+    public async writeStruct(clusterKey: number | string, payload: KeyValue, options?: Options): Promise<void> {
+        const cluster = Zcl.Utils.getCluster(clusterKey);
+        options = this.getOptionsWithDefaults(
+            options, true, Zcl.Direction.CLIENT_TO_SERVER, cluster.manufacturerCode);
+
+        const frame = Zcl.ZclFrame.create(
+            Zcl.FrameType.GLOBAL, options.direction, options.disableDefaultResponse,
+            options.manufacturerCode, options.transactionSequenceNumber ?? ZclTransactionSequenceNumber.next(),
+            `writeStruct`, cluster.ID, payload, options.reservedBits
+        );
+
+        const log = `WriteStruct ${this.deviceIeeeAddress}/${this.ID} ` +
+            `${cluster.name}(${JSON.stringify(payload)}, ${JSON.stringify(options)})`;
+        debug.info(log);
+
+        try {
+            await Entity.adapter.sendZclFrameToEndpoint(
+                this.deviceIeeeAddress, this.deviceNetworkAddress, this.ID, frame, options.timeout,
+                options.disableResponse, options.disableRecovery, options.srcEndpoint
+            );
```

Comment:
In the ZCL I read there is a "Write Attributes Structured Response Command", does your device send this back on write?

Reply:
Although I guess there's very little that one can do with a "Write Attributes Structured Response" anyway, as with "Write Attributes Response", beyond letting the user know that the action failed, under the assumption that the attributes always exist and are writable :shrug:

Reply:
> The response is quite exotic and simply acknowledges the attribute and value that has been set.

What response do you get?

Reply:
I reviewed the PR, no comments besides this one. Still wondering if we need something like https://github.com/Koenkk/zigbee-herdsman/blob/00f91b303a7e49d6763de6d2033a20baee15ecd9/src/controller/model/endpoint.ts#L304 here.

Reply:
I wouldn't know what to put in there that wouldn't break the tests or coverage, as at that point the value of `Entity.adapter.sendZclFrameToEndpoint` is `null`, so I can't simply do this:

```ts
if (!options.disableResponse) {
    this.checkStatus(result.frame.Payload);
}
```

I can either leave it as it is, or go all in and implement the correct response path (decoding of `WriteStructuredResponse`). What do you think is best?

Reply:
If your device sends a `WriteStructuredResponse` back please add it, if not add `// TODO`

Reply:
I simply added the `// TODO`. Thanks!